File: backend/src/utils/dto/register.dto.ts
import { IsEmail, IsNotEmpty, Length } from 'class-validator';
import { MAX_PASSWORD_LENGTH, MIN_PASSWORD_LENGTH } from '../constants';

class RegisterDTO {
  @IsEmail()
  email: string;

  @Length(MIN_PASSWORD_LENGTH, MAX_PASSWORD_LENGTH)
  password: string;

  @IsNotEmpty()
  username: string;
}

export default RegisterDTO;

File: Android/app/src/main/java/com/example/pizzeria/domain/RegisterDTO.java
package com.example.pizzeria.domain;

public class RegisterDTO {
    public String username;
    public String email;
    public String password;

    public RegisterDTO(String username, String email, String password) {
        this.username = username;
        this.email = email;
        this.password = password;
    }
}
